File: src/Binacle.Net.Lib/Fitting/Algorithms/First Fit Decreasing v1/Bin.cs
﻿using Binacle.Net.Lib.Abstractions.Models;

namespace Binacle.Net.Lib.Fitting.Algorithms;

internal sealed partial class FirstFitDecreasing_v1
{
	private sealed class Bin : VolumetricItem, IWithID
	{
		public Bin(string id, IWithReadOnlyDimensions item) :
			base(item)
		{
			this.ID = id;
		}

		public Bin(string id, int length, int width, int height) :
			base(length, width, height)
		{
			this.ID = id;
		}

		public string ID { get; set; }
	}
}

File: web/_js/models/bin.js
import {ValidatableBox} from "./validatableBox";

class Bin extends ValidatableBox {
	constructor(length, width, height) {
		super(length, width, height);
	}

	get id() {
		return `${this.length}x${this.width}x${this.height}`;
	}
}

export default Bin;
